### DISCUSSION TITLE
Belgium, Flanders: Ecopower

### DISCUSSION THREAD

**Discussion by lennie-BE** [JaccoR/hass-entso-e#145] [Price Modifyer Templates]
For the variable tariffs from Ecopower the templates below should work. I've checked these with Ecopower support.

Note! check the values of the parameters on the most recent tariff cards here: https://www.ecopower.be/groene-stroom/archief-tariefkaarten

Some variables are depending on the region, the type of contract, or even the total amount af annual consumption. So check it out!

**Template for consumption:**

```
{% set s = {
"bijz_accijns": 0.04748,
"energiebijdrage": 0.0019261,
"kost_gsc": 0.01581,
"kost_wkk": 0.002576,
"afnametarief": 0.0508125,
"btw": 1.06,
"marge_procentueel": 1.02,
"marge_per_kwh": 0.004
} %}
{{((((current_price * s.marge_procentueel) + s.marge_per_kwh) + s.bijz_accijns + s.energiebijdrage + s.kost_gsc + s.kost_wkk + s.afnametarief) * s.btw) | float}}
```

**Template for injection:**

```
{% set s = {
"marge_procentueel": 0.85,
"marge_per_kwh": 0.004
} %}
{{((current_price * s.marge_procentueel) - s.marge_per_kwh) | float}}
```

**Comment by willembuys** [comment]
Ecopower has updated their price formula for injection, which is now 98% of the day-ahead price with a fixed margin of 15 euro/MWh.
If I update the values in the above template for injection, this results in:

```
 {% set s = {
"marge_procentueel": 0.98,
"marge_per_kwh": 0.015
} %}
{{((current_price * s.marge_procentueel) - s.marge_per_kwh) | float}}
```

**Reply by JohanDeckx** [reply]
Being quite new to HA, I can't find where to integrate the INJECTION template???
Anyone has some suggestion to me?

Thanks

**Reply by willembuys** [reply]
I have setup two devices: one for consumption, one for injection.

**Reply by EnigmaTD** [reply]
did you use the same token for both devices?
I did not save my token, can it be retrieved somewhere?

**Reply by willembuys** [reply]
Yes, both devices use the same API key.
It is stored in the configuration of a device.
If you don't have your token anymore, you can create a new one on the ENTSO-E platform.

**Comment by dagobert00** [comment]
Hi. How do you incorporate 'fixed' costs ? For example: capacity tariff and subscription cost in the template to reflect the full gross electricity price in HomeAssistant?

**Reply by willembuys** [reply]
The template only allows to calculate the actual hourly kWh-tariff. I use the integration 'Dynamic energy cost' to calculate the consumption total, injection total and self-consumption value.
Then I manually sum up consumption total, injection total (minus), subscription cost and capacity tariff (kW rate x monthly peak) to calculate the monthly cost.
The capacity tariff is a simplification, as your invoice is calculated on the 12-month moving average peak.

**Reply by dagobert00** [reply]
Hi. Thanks for your reply. I understand you want to leverage the dynamic energy cost integration to track cost of individual contributors but for the overall cost I don't see the benefit compared to the default Energy dashboard ? What do I miss?

**Reply by willembuys** [reply]
I haven't found a way to use the values in the energy dashboard in own calculations, that's why I calculate them separately. It provides flexibility.

**Reply by EnigmaTD** [reply]
> The template only allows to calculate the actual hourly kWh-tariff. I use the integration 'Dynamic energy cost' to calculate the consumption total, injection total and self-consumption value. Then I manually sum up consumption total, injection total (minus), subscription cost and capacity tariff (kW rate x monthly peak) to calculate the monthly cost. The capacity tariff is a simplification, as your invoice is calculated on the 12-month moving average peak.

How do you identify the monthly peak (in order to calculate the capacity tariff)?

**Reply by willembuys** [reply]
My P1-dongle reports the monthly peak.
